### PR TITLE
Add clean shutdown data consistency test

### PR DESCRIPTION
**Pull request by katlapinka**
Signed-off-by: Katarzyna Lapinska <katarzyna.lapinska@intel.com>

### CODE REVIEW

File: test/functional/tests/fault_injection/test_reboot.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+mount_point = "/mnt/test"
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.parametrize("reboot_type", ["soft", "hard"])
+@pytest.mark.require_plugin("power_control")
+def test_planned_reboot(reboot_type, cache_mode):
```
Comment (mmichal10) [inline test/functional/tests/fault_injection/test_reboot.py:26]: Did you skip test description and pass criteria on purpose?

File: test/functional/tests/fault_injection/test_reboot.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            power_control = TestRun.plugin_manager.get_plugin('power_control')
+            power_control.power_cycle()
+
+    with TestRun.step("Load cache."):
+        first_byte_in_octal = TestRun.executor.run(
+            f"hexdump -b -n 1 {cache_dev.system_path}").stdout.split("\n")[0]
+        first_byte_in_octal = first_byte_in_octal.split()[1]
+        if first_byte_in_octal != "001":
+            TestRun.LOGGER.error(f"CAS dirty block is incorrect. Is {first_byte_in_octal},"
+                                 f" should be 001")
```
Comment (mmichal10) [inline test/functional/tests/fault_injection/test_reboot.py:59]: Do you know if it this check is necessary?

File: test/functional/tests/fault_injection/test_reboot.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.parametrize("reboot_type", ["soft", "hard"])
+@pytest.mark.require_plugin("power_control")
+def test_planned_reboot(reboot_type, cache_mode):
+    with TestRun.step("Prepare CAS device."):
+        cache_dev = TestRun.disks['cache']
+        core_dev = TestRun.disks['core']
+        cache = casadm.start_cache(cache_dev, cache_mode)
+        core = cache.add_core(core_dev)
+        core.create_filesystem(Filesystem.ext4, blocksize=int(Size(1, Unit.Blocks4096).get_value()))
```
Comment (robertbaldyga) [inline test/functional/tests/fault_injection/test_reboot.py:32]: You don't need to call `get_value()` method here. It's called automatically when casting object of `Size` to `int`.

File: test/functional/tests/data_integrity/test_reboot.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            power_control.power_cycle()
+
+    with TestRun.step("Load cache."):
+        cache = casadm.load_cache(cache_dev)
+        core.mount(mount_point)
+
+    with TestRun.step("Check file md5sum."):
+        test_file.refresh_item()
+        if test_file_md5 != test_file.md5sum():
+            TestRun.fail("Checksums does not match - file is corrupted.")
```
Comment (Ostrokrzew) [inline test/functional/tests/data_integrity/test_reboot.py:70]: You can use `TestRun.LOGGER.error` to let test remove test file.

File: test/functional/tests/initialize/test_clean_reboot.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+        core_dev = TestRun.disks['core']
+        cache = casadm.start_cache(cache_dev, cache_mode)
+        core = cache.add_core(core_dev)
+        core.create_filesystem(filesystem, blocksize=int(Size(1, Unit.Blocks4096)))
+        core.mount(mount_point)
+
+    with TestRun.step("Create file on cache and count it's checksum."):
```
Comment (Deixx) [inline test/functional/tests/initialize/test_clean_reboot.py:45]: its

File: test/functional/tests/initialize/test_clean_reboot.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        test_file.refresh_item()
+        if test_file_md5 != test_file.md5sum():
+            TestRun.LOGGER.error("Checksums does not match - file is corrupted.")
+        else:
+            TestRun.LOGGER.info("File checksum is correct.")
+
+    with TestRun.step("Remove test file and stop cache."):
+        test_file.remove()
+        core.unmount()
+        cache.stop()
```
Comment (Deixx) [inline test/functional/tests/initialize/test_clean_reboot.py:77]: Just remove file - cleanup should take care of the rest

**Review by Ostrokrzew — APPROVED** [review]

**Review by imjfckm — APPROVED** [review]

**Pull request merged**